processors/events_dashboard_processor.py

```diff
@@ -0,0 +1,179 @@
+    timestamp_utc = datetime.fromtimestamp(timestamp_seconds, timezone.utc)
+    return timestamp_utc
+
+
+def process_events(event_data, event_type):
+    event_timestamp_utc = timestamp_utc_conversion(event_data["timestamp"])
+    user_uuid_md5 = hashlib.md5(
+        event_data["user_uuid"].encode("utf-8")
+        ).hexdigest()
+    modified_event_data = {
+        "user_uuid_md5": user_uuid_md5,
+        "course_id": event_data["course_id"],
+        "impression_id": event_data["impression_id"],
+        "timestamp": event_timestamp_utc,
+        "content_id": event_data["content_id"],
+        "variant": event_data["variant"],
+    }
+
+    if event_type == "input_submitted_event":
+        modified_event_data["input_content_id"] = event_data[
+            "input_content_id"
+        ]
+    elif event_type == "pset_problem_attempted_event":
+        modified_event_data["pset_content_id"] = event_data[
+            "pset_content_id"
+        ]
+        modified_event_data["pset_problem_content_id"] = event_data[
+            "pset_problem_content_id"
+        ]
+        modified_event_data["problem_type"] = event_data["problem_type"]
+        modified_event_data["correct"] = event_data["correct"]
+        modified_event_data["attempt"] = event_data["attempt"]
+        modified_event_data["final_attempt"] = event_data["final_attempt"]
+
+    with session_factory() as session:
+        if event_type == "content_loaded_event":
+            session.add(ContentLoadedEvent(**modified_event_data))
+        elif event_type == "input_submitted_event":
+            session.add(InputSubmittedEvent(**modified_event_data))
+        elif event_type == "pset_problem_attempted_event":
+            session.add(PsetProblemAttemptedEvent(**modified_event_data))
+
+        commit_ignoring_unique_violations(session)
+
+
+def process_s3_notification(s3_client, s3_notification, event_type):
+    for record in s3_notification["Records"]:
+        event_name = record["eventName"]
+
+        if not event_name.startswith("ObjectCreated:"):  # pragma: no cover
+            raise ProcessorException(f"Unexpected S3 event: {event_name}")
+
+        s3_data = record["s3"]
+        bucket = s3_data["bucket"]["name"]
+        key = unquote(s3_data["object"]["key"])
+
+        event_data = s3_client.get_object(Bucket=bucket, Key=key)
+
+        types_of_events = {
+            "content_loaded_event": "content_loaded_event",
+            "input_submitted_event": "input_submitted_event",
+            "pset_problem_attempted_event": "pset_problem_attempted_event",
+        }
+
+        avro_reader = reader(event_data["Body"])
+        for event in avro_reader:
+            for field in [
+                "source_scheme",
+                "source_host",
+                "source_path",
+                "source_query",
+            ]:
+                del event[field]
+            if event_type != "content_loaded_event":
+                del event["response"]
```

Comment:
Is any of this necessary? I think you're only pulling desired fields out in `process_events()`, so I think the field deletion logic here can be removed w/o impact :man_shrugging: .

Reply:
It is not necessary - I copied this pattern over from `events_enclave_processor.py`. I now see that in the aforementioned file we didn't pass all the event data to another function and then pull out the desired fields, but we instead remove unwanted fields via the for loop. I've removed the loop and associated conditional.